### PR TITLE
Feature html client

### PR DESCRIPTION
**Pull request by playamongthestars**

### CODE REVIEW

File: classes/class.ilAdobeConnectXMLAPI.php [hunk 1]
```diff
@@ -247,6 +249,45 @@ public function logout($session)
...
+
+		$context    = array(
+			'http'  => array('timeout' => 4),
+			'https' => array('timeout' => 4)
+		);
+		$ctx        = $this->proxy($context);
+		$xml_string = file_get_contents($url, false, $ctx);
+		$xml        = simplexml_load_string($xml_string);
+
+		if($xml && $xml->common->cookie != "")
```
Comment (mjansenDatabay) [inline classes/class.ilAdobeConnectXMLAPI.php]: Is this (`$xml->common->cookie`) really required?

Comment (playamongthestars) [reply classes/class.ilAdobeConnectXMLAPI.php]: no ;-)

Comment (playamongthestars) [reply classes/class.ilAdobeConnectXMLAPI.php]: fixed

File: classes/class.ilAdobeConnectServer.php [hunk 2]
```diff
@@ -92,6 +92,15 @@ class ilAdobeConnectServer
 	 * @var string $x_user_id Header-Variable needed for HTTP-Header-Auth.
 	 */
 	private $x_user_id = null;
+
+	/**
+	 * @var bool
+	 */
+	private $html_client = false;
```
Comment (mjansenDatabay) [inline classes/class.ilAdobeConnectServer.php:99]: Should be named `$use_html_client`

File: classes/class.ilAdobeConnectServer.php [hunk 3]
```diff
@@ -341,6 +353,38 @@ public function getXUserId()
...
+	 */
+	public function setApiVersion($api_version)
+	{
+		$this->api_version = $api_version;
+	}
+
+	/**
+	 * @return bool
+	 */
+	public function getHtmlClient()
```
Comment (mjansenDatabay) [inline classes/class.ilAdobeConnectServer.php]: `isHtmlClientEnabled`, or `shouldPreferHtmlClient`,  or `shouldUseHtmlClient`, etc.

File: classes/class.ilAdobeConnectServer.php [hunk 4]
```diff
@@ -341,6 +353,38 @@ public function getXUserId()
...
+	{
+		return $this->html_client;
+	}
+
+	/**
+	 * @param bool $html_client
+	 */
+	public function setHtmlClient($html_client)
```
Comment (mjansenDatabay) [inline classes/class.ilAdobeConnectServer.php]: `setHtmlClientUsage` or something similar, which indicates that a boolean value could be set with this mutator

File: classes/class.ilObjAdobeConnect.php [hunk 5]
```diff
@@ -1765,4 +1777,20 @@ public function setAcLanguage($ac_language)
 	{
 		$this->ac_language = strtolower($ac_language);
 	}
+
+	/**
+	 * @return bool
+	 */
+	public function getHtmlClient()
```
Comment (mjansenDatabay) [inline classes/class.ilObjAdobeConnect.php]: Please change the name of accessor and mutator (same as in the plugin administration).

File: classes/class.ilAdobeConnectConfigGUI.php [hunk 6]
```diff
@@ -672,6 +673,33 @@ public function initIliasSettingsForm()
...
+		}
+		$this->form->addItem($html_client);
+	}
+
+	/**
+	 * 
+	 */
+	private function readApiVersion()
+	{
+		$xmlAPI = ilXMLApiFactory::getApiByAuthMode();
```
Comment (mjansenDatabay) [inline classes/class.ilAdobeConnectConfigGUI.php:695]: Does this also work even if the plugin configuration is not complete? We should prevent an issue/error if the settings view is accessed by the user, but the configuration is not complete, yet. The settings should not be displayed for an incomplete configuration.

**Pull request merged**